subsys/bluetooth/audio/bap_broadcast_sink.c

```diff
@@ -649,46 +649,51 @@ static bool pa_decode_base(struct bt_data *data, void *user_data)
 	const struct bt_bap_base *base = bt_bap_base_get_base_from_ad(data);
 	struct bt_bap_broadcast_sink_cb *listener;
 	int base_size;
-	int ret;
 
 	/* Base is NULL if the data does not contain a valid BASE */
 	if (base == NULL) {
 		return true;
 	}
 
-	if (atomic_test_bit(sink->flags, BT_BAP_BROADCAST_SINK_FLAG_BIGINFO_RECEIVED)) {
-		ret = base_get_bis_count(base);
-
-		if (ret < 0) {
-			LOG_DBG("Invalid BASE: %d", ret);
-			return false;
-		} else if (ret != sink->biginfo_num_bis) {
-			LOG_DBG("BASE contains different amount of BIS (%u) than reported by "
-				"BIGInfo (%u)",
-				ret, sink->biginfo_num_bis);
-			return false;
-		}
-	}
-
 	/* We provide the BASE without the service data UUID */
 	base_size = bt_bap_base_get_size(base);
+	if (base_size != sink->base_size || memcmp(base, sink->base, base_size) != 0) {
```

Comment:
Where is `sink->base_size` assigned before getting here?

Reply:
The value will remain `0` until we receive a BASE, which we get from a PA sync. 
On `bt_disable` the PA syncs will be terminated, and `pa_term_cb` will be called, resetting the value